### PR TITLE
fix(workspace): bound table height to terminal viewport for large result sets

### PR DESCRIPTION
**Pull request by fxenik**
## 🔗 Ticket

[DEX-266](https://linear.app/rudderstack/issue/DEX-266)

---

## Summary

Running `rudder-cli workspace accounts list` with >80 accounts caused the TUI to render poorly because the table height was set to `len(rows)+1`, rendering every row at once with no viewport constraint. This fix caps the table to a bounded default height and dynamically resizes on terminal window changes, enabling bubbletea's built-in scrolling.

---

## Changes

* Set a default table height of 20 instead of unbounded `len(rows)+1`
* Recalculate table height dynamically on `WindowSizeMsg`, reserving space for help footer
* Constrain the details pane height to match the table using `lipgloss.Height()` on the rendered table view, avoiding off-by-one errors from manual header/border offset calculations
* Add unit tests verifying height is bounded and minimum height is enforced

---

## Testing

* Unit tests added (`cli/internal/lister/table_test.go`)
* `make lint` — passes
* `make test` — passes

---

## Risk / Impact

Low — only affects TUI table rendering; no changes to data flow or business logic.

---

## Checklist

* [x] Ticket linked
* [x] Tests added/updated
* [x] No breaking changes (or documented)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are limited to TUI sizing/rendering and add unit coverage; no data fetching or business logic is modified.
> 
> **Overview**
> Fixes `workspace accounts list` TUI rendering for large result sets by **capping and dynamically resizing** the table height to the terminal viewport (reserving space for the help footer and enforcing a minimum height).
> 
> Updates the layout to **match the details pane height to the rendered table height** (via `lipgloss.Height`) and adds unit tests covering window resize height calculation and minimum-height behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ab2eee0000bdf54b09c9e13282369111f57663d2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: cli/internal/lister/table.go [hunk 1]
```diff
@@ -88,9 +93,11 @@ func (m model) View() string {
 	detailsContent := lipgloss.NewStyle().Padding(0, 2).Render(detailsView)
 	fullDetailsView := lipgloss.JoinVertical(lipgloss.Top, detailsHeader, ruler, detailsContent)
 
-	// Main Layout
+	// Main Layout — constrain details pane to match table height
+	tableHeight := m.table.Height() + 1 // +1 for header border
 	detailsStyle := lipgloss.NewStyle().
-		Padding(0, 2)
+		Padding(0, 2).
+		Height(tableHeight)
```
[the remark this answers is not in the document]
Comment (fxenik) [reply cli/internal/lister/table.go]: **Claude:** Here's my plan for this:

**What:** Fix off-by-one in details pane height by measuring actual rendered table height instead of manual offset.

**How:**
- Replace `m.table.Height() + 1` with `lipgloss.Height(m.table.View())` in `cli/internal/lister/table.go`
- Reuse the rendered table view string to avoid calling `View()` twice

---
Implemented in commit b2e876d2. Uses `lipgloss.Height()` on the rendered table view to get the correct height regardless of header border style, eliminating the off-by-one.
